profiles.wordpress.org/public_html/wp-content/plugins/wporg-5ftf-activity-handler/wporg-5ftf-activity-handler.php

```diff
@@ -0,0 +1,113 @@
+<?php
+/*
+Plugin Name: WordPress.org 5ftf Activity Handler
+Plugin URI: http://wordpress.org
+License: GPL2
+Version: 1.1
+Description: Handles saving of last 5ftf contribution.
+*/
+
+defined( 'ABSPATH' ) or die();
+
+if ( ! class_exists( 'WPOrg_5ftf_Activity_Handler' ) ) {
+
+	class WPOrg_5ftf_Activity_Handler {
```

Comment:
What's the benefit of using a class in this case? It doesn't seem like it's using any of the distinctive features of OOP like inheritance, polymorphism, encapsulation, etc.

Reply:
Fair question. :)

I originally wanted to pass in wp dependencies so we could write tests that don't touch the database but stopped short. I'm not convinced I can do that in a way that doesn't equal a huge constructor with too many optional parameters filled with wp objects and wp functions. Additionally, reconstructing mock objects or finding drop-in replacements for this specific plugin is probably overkill.

I don't think an OOP approach introduces any future problems and I think it still leaves the door open to more abstracted code in the future.

Reply:
Yeah, testing in WP is not ideal :/

I lean more towards YAGNI and lean/straight-forward code where OOP is only used when it solves a tangible problem, but this is a valid approach too 👍🏻